vite.config.ts

```diff
@@ -1,24 +1,74 @@
 import react from "@vitejs/plugin-react-swc";
-import { defineConfig } from "vite";
+import { defineConfig, loadEnv } from "vite";
 import svgr from "vite-plugin-svgr";
 import tsconfigPaths from "vite-tsconfig-paths";
+import prerender from "@prerenderer/rollup-plugin";
+import axios from "axios";
+
+type Performance = {
+  performanceId: number;
+  performanceTitle: string;
+  performancePeriod: string;
+  ticketPrice: number;
+  dueDate: number;
+  genre: string;
+  posterImage: string;
+  performanceVenue: string;
+};
+
+// 사전 렌더링 경로 생성
+async function generatePerformanceRoutes(url: string) {
+  const staticRoutes = ["/", "/main"]; // 정적 경로
+  const dynamicRoutes = await fetchPerformanceIds(url); // 동적 경로 가져오기
+
+  return staticRoutes.concat(dynamicRoutes);
+}
+
+// 공연 ID를 가져오는 API 호출
+async function fetchPerformanceIds(url: string) {
+  try {
+    const response = await axios.get(`${url}/main`);
+
+    // 응답 데이터에서 performanceId 추출
+    const performances = response.data.data.performanceList;
+    const validIds = performances.map((performance: Performance) => performance.performanceId);
+
+    return validIds.map((id: number) => `/gig/${id}`);
```

Comment:
p3) 불필요한 map이 반복되는 거 같아서 하나로 바꿀게요 ~
```suggestion
 const validIds = performances.map(
      (performance: PerformanceType) => `/gig/${performance.performanceId}`
    );

return validIds;
```

Reply:
헤헤헤헤ㅔㅎ 레전드 개발자 정도영 믿고 있었다구~